website/photos/views.py

```diff
@@ -132,9 +137,11 @@ class LikedPhotoView(LoginRequiredMixin, PagedView):
     context_object_name = "photos"
 
     def get_queryset(self):
```

Comment:
In a few places the change is in `get_queryset`. Does that work? I guess as long as the returned queryset is not modified outside of get_queryset it would work on the cached contents, but if it changes at all it would still do N+1 afterwards

Reply:
I tested using the Django debug toolbar and it seems to work. I agree that it's not a perfect solution though, maybe if there was something similar to `prefetch_related` for custom code?

Reply:
Yeah there isn't... If it had all been nicely made with models instead non-model python objects it would be possible with a Prefetch object, but alas
So I guess this is still the best